docs/our-approach-to-data.md

```diff
@@ -0,0 +1,267 @@
+
+Framed this way, we can consider two types of container components: connected components and fetching components.
+
+#### Connected components
+
+Separating visual and data concerns is a good mindset to have when approaching components, and whenever possible, we should strive to create reusable visual components which accept simple props for rendering. However, pragmatically it is unreasonable to assume that components will always be reused and that there's always a clear divide between the visual and data elements. As such, while we recommend creating purely visual components whenever possible, it is also reasonable to create components that are directly tied to the global application state. We call these "connected" components, and we use the [`react-redux` library](https://github.com/rackt/react-redux) to assist in creating bindings between React components and the store instance.
+
+Below is an example of a connected component using [`react-redux`'s `connect`](https://github.com/rackt/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) function. It retrieves an array of posts for a given site and passes the posts to the component for rendering. If you're unfamiliar with the stateless function syntax for declaring components, refer to the [React 0.14 upgrade guide](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#stateless-functional-components) for more information.
+
+```jsx
+function PostsList( { posts } ) {
+	return (
+		<ul className="posts-list">
+			{ posts.map( ( post ) => {
+				return <li key={ post.id }>{ post.title }</li>;
+			} ) }
+		</ul>
+	);
+}
+
+export default connect( ( state, ownProps ) => {
+	return {
+		posts: getSitePosts( ownProps.siteId )
+	};
+} )( PostsList );
+```
+
+The `connect` function accepts two arguments, and they each serve a distinct purpose. Both pass props to the connected component, and are respectively used to provide data and handle behavior on behalf of the component.
+
+1. `mapStateToProps`: A function which, given the store state, returns props to be passed to the connected component. This is used to satisfy the need to make data available to the component.
+2. `mapDispatchToProps`: A function which, given the store dispatch method, returns props to be passed to the connected component. This is used to satisfy the need to allow the component to update the store state.
+
+As an example, consider a component which renders a Delete button for a given post. We want to display the post title as a label in the delete button, and allow the component to trigger the post deletion when clicked.
+
+```jsx
+function PostDeleteButton( { label, delete } ) {
+	return (
+		<button onClick={ delete }>
+			{ this.translate( 'Delete "%s"', { args: [ label ] } ) }
```

Comment:
Is `this.translate()` available to stateless function components? I didn't think it was.